zebrad/src/commands/start/sync.rs

```diff
@@ -1,34 +1,37 @@
 use color_eyre::eyre::{eyre, Report};
 use futures::stream::{FuturesUnordered, StreamExt};
-use std::{collections::HashSet, iter, time::Duration};
+use std::{collections::HashSet, iter, sync::Arc, time::Duration};
 use tokio::time::delay_for;
 use tower::{Service, ServiceExt};
 use tracing_futures::Instrument;
-use zebra_chain::{block::BlockHeaderHash, types::BlockHeight};
+use zebra_chain::{
+    block::{Block, BlockHeaderHash},
+    types::BlockHeight,
+};
 
 use zebra_network as zn;
 use zebra_state as zs;
 
-pub struct Syncer<ZN, ZS>
+pub struct Syncer<ZN, ZS, ZC>
 where
     ZN: Service<zn::Request>,
 {
     pub peer_set: ZN,
-    // TODO(jlusby): add validator
     pub state: ZS,
+    pub verifier: ZC,
```

Comment:
Since it's a verifier service I think that `ZV` would be a slightly preferable name to `ZC`; presumably "C" is short for "consensus" but I think that "verifier" is closer to the actual role of the object (& "V" stands for fewer things than "C").